CDP4Dal/Session.cs

```diff
@@ -835,16 +889,19 @@ private async Task Update(Thing thing, bool isRefresh = true)
             var queryAttribute = new DalQueryAttributes { RevisionNumber = revisionNumber };
 
             // Create the token source
-            this.cancellationTokenSource = new CancellationTokenSource();
+            var cancellationTokenSource = new CancellationTokenSource();
+            var cancellationTokenKey = Guid.NewGuid();
+            this.cancellationTokenSourceDictionary.TryAdd(cancellationTokenKey, cancellationTokenSource);
             IEnumerable<CDP4Common.DTO.Thing> dtoThings;
 
             try
             {
-                dtoThings = await this.Dal.Read(thing.ToDto(), this.cancellationTokenSource.Token, queryAttribute);
+                dtoThings = await this.Dal.Read(thing.ToDto(), cancellationTokenSource.Token, queryAttribute);
+                this.cancellationTokenSourceDictionary.TryRemove(cancellationTokenKey, out cancellationTokenSource);
             }

```

Comment:
please use the same pattern as in the `Read` method where we have a 
```
try catch finally
```

where in the finally we should have the `TryRemove` on the `cancellationTokenSourceDictionary` and i would prefer syntax is the same all over and explicit and easy to read.